_RELEASE/Packs/base/Scripts/utils.lua

```diff
@@ -539,10 +557,84 @@ function ArrayRemoveIf(t, fnRemove)
     return t
 end
 
+-- Tests whether a table contains a specific value on any existing key
+function tableContainsValue(val, table)
+    for _, v in pairs(table) do
+        if val == v then return true end
+    end
+    return false
+end
+
 function getBPMToBeatPulseDelay(bpm)
     return 3600 / bpm
 end
 
 function getMusicDMSyncFactor()
     return u_getDifficultyMult() ^ 0.12
 end
+
+-- Sets hue to a specific value by setting its min an max to the same value
+function forceSetHue(h)
+    s_setHueMin(h)
+    s_setHueMax(h)
+end
+
+-- Sets pulse to a specific value by setting its min an max to the same value
+function forceSetPulse(p)
+    s_setPulseMin(p)
+    s_setPulseMax(p)
+end
+
+-- Guarantees an input value to be a valid number of sides. Falls back to the level's current number of sides if an invalid argument is given
+function verifyShape(shape)
+    return type(shape) == 'number' and math.floor(math.max(shape, 3)) or l_getSides()
+end
+
+-- Distance from the center to the player position
+function getDistanceBetweenCenterAndPlayer()
+    return l_getRadiusMin() * l_getPulse() / l_getPulseMin() + l_getBeatPulse()
+end
+-- Distance from center to tip of player arrow
+function getDistanceBetweenCenterAndPlayerTip()
+    return getPlayerRadius() + 7.25
```

Comment:
The tip of the triangle is 7.3 in the engine code. I would recommend adding a constant above instead of hardcoding the value here and in the function below